### PR TITLE
feat(selectNature): display nature bonus and maluses

### LINKED ISSUE #46: Display nature bonuses and maluses when relevant
As a **User**
I would like to **know what are nature bonuses and maluses from the nature dropdown lists**
So that **I know what are thoses bonuses and maluses without needing to check on another website**

## Acceptance criterias
- The statistic modifications are displayed when the user is hovering a value of the dropdown list

> [!NOTE]
> See the [Flo's suggestion on Discord](https://discord.com/channels/143824995867557888/1141994345793331240/1141994345793331240).

### PR DESCRIPTION
**Pull request by Menthx**
Display nature bonus and maluses on Select associeted.
 
Thank you for your contribution to the **Pokémon Studio** repo.

Before submitting this PR into the develop branch, please make sure:

- [x ] Your code builds clean without any errors or warnings
- [ x] You are following the [Code guidelines](../CodeGuidelines.md)
- [ x] You tested your code to make sure it does what it is supposed to do

## Description

Add some indicator into the slect nature for Pokemon to display some stats associated !

## Tests to perform

Go into select nature component thought Battler page or Group page and test it !

Closes: https://github.com/PokemonWorkshop/PokemonStudio/issues/46

### CODE REVIEW

File: src/views/components/selects/SelectNature.tsx [hunk 1]
```diff
@@ -1,30 +1,105 @@
...
+
+/**
+ * Get the nature options
+ * @param state
+ * @param t useTranslation
+ * @returns SelectOption[]
+ */
+const getNatureOptions = (state: State, t: TFunction<['database_natures']>): SelectOption[] =>
+  Object.entries(state.projectConfig.natures.db_symbol_to_id).map(([value, natureId]) => {
+    const test = getNatureExtraStatsComparedToDependingStats(state, t);
```
Comment (NuriYuri) [inline src/views/components/selects/SelectNature.tsx]: You should not call this function on each iteration as it's an expensive one and it does not depend on any parameters of the iteration.

Add curly brackets to getNatureOptions and return the Object.entries :)

**Comment by andykimpe** [comment]
as I say [here](https://github.com/PokemonWorkshop/PokemonStudio/issues/312) when have added text in a language
all variables must be answered in all languages proposed by the program
either translate the other languages
false report variables in English in other languages

[Compare](https://github.com/andykimpe/PokemonStudio/compare/294e13fb08d30036f07001c75934ee201307c891...andykimpe:PokemonStudio:02f2596f743a64ae2468a6e36fbeb866066ae933)

**Comment by AerunDev** [comment]
> as I say [here](https://github.com/PokemonWorkshop/PokemonStudio/issues/312) when have added text in a language all variables must be answered in all languages proposed by the program either translate the other languages false report variables in English in other languages
> 
> [Compare](https://github.com/andykimpe/PokemonStudio/compare/294e13fb08d30036f07001c75934ee201307c891...andykimpe:PokemonStudio:02f2596f743a64ae2468a6e36fbeb866066ae933)

Rey & Palb already answered on the other post. If you want to contribute, feel free to create some PR to add the translation if you are fluent in the related languages. Translations via translators without peer review will not be accepted.

File: src/views/components/selects/SelectNature.tsx (hunk 1 above)
Comment (Menthx) [reply src/views/components/selects/SelectNature.tsx]: Ah yes, did'nt see i was iterate multiple time on it, thanks for review !

**Review by AerunDev — APPROVED** [review]
Testé et approuvé, bien joué ! 😄

**Review by NuriYuri — APPROVED** [review]

**Pull request merged**